packages/react-reconciler/src/ReactFiberBeginWork.new.js

```diff
@@ -2661,18 +2662,32 @@ function updateDehydratedSuspenseComponent(
     // This boundary is in a permanent fallback state. In this case, we'll never
     // get an update and we'll never be able to hydrate the final content. Let's just try the
     // client side render instead.
+    const {
+      errorMessage,
+      errorComponentStack,
+      errorHash,
+    } = getSuspenseInstanceFallbackErrorDetails(suspenseInstance);
+    const error = errorMessage
+      ? // eslint-disable-next-line react-internal/prod-error-codes
+        new Error(errorMessage)
+      : new Error(
+          'The server could not finish this Suspense boundary, likely ' +
+            'due to an error during server rendering. Switched to ' +
+            'client rendering.',
+        );
+    if (errorHash) {
+      (error: any).hash = errorHash;
+    }
+    if (__DEV__) {
+      if (errorComponentStack) {
+        (error: any).componentStack = errorComponentStack;
```

Comment:
We should not add the componentStack this way since we already have a separate protocol for that which is errorInfo.

In this first PR you can just exclude it.

Now that I think about it, it probably makes the most sense to do the same with the hash.

Regardless, let's not expose anything extra to users until we're done with the API.

Reply:
roger